File: Data Structures and Algorithms/Sorting/Selection_Sort.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n; cin >> n;
    vector <int> v(n);
    for(int i = 0; i < n; i++){
        cin >> v[i];
    }

    // * Implementation of Selection Sort
    
    for(int i = 0; i < n; i++){
        int minIndex = i;
        for(int j = i + 1; j < n; j++){
            if(v[j] < v[minIndex]){
                minIndex = j;
            }
        }
        swap(v[i], v[minIndex]);
    }


    for(int i = 0; i < n; i++){
        cout << v[i] << " ";
    }
    cout << endl;
    return 0;
}

File: Data Structures and Algorithms/Sorting/Selection Sort.py
from psutil import swap_memory


n = int(input())
v = list(map(int,input().strip().split()))[:n]

# *Implementation of Selection Sort

for i in range(n):
    minIndex = i
    for j in range(i + 1, n - 1):
        if v[i] < v[minIndex]:
            minIndex = j
    v[i], v[minIndex] = v[minIndex], v[i]

for i in range(n):
    print(v[i])